lib/Arch/Runtime/HyperCall.cpp

```diff
@@ -44,6 +44,20 @@
 
 Memory *__remill_sync_hyper_call(State &state, Memory *mem,
                                  SyncHyperCall::Name call) {
+
+#if REMILL_HYPERCALL_X86 || REMILL_HYPERCALL_AMD64
```

Comment:
I think for x86, you can't use `r`-prefixes, you need the `e`-prefixes, e.g. `ebp` vs. `rbp`. I think you'll need different variants of this per architecture. So on amd64, have the 64-bit variants, that touch `.qword`, and on x86, have the 32-bit variants that touch `.dword`.

I am confused as to why these are all pointers into the state structure. As far as I can tell, you are saying: this register must hold the address of this field in the state structure, as opposed to: this register must hold the value of the field in the state structure.

I am slightly sketched out about the implicit stack switch here with `rsp`. I don't think we can properly express that. Otoh, it doesn't really matter, because we're trying to express the dataflow, not something that is practically compilable, so it's fine.

In the register constraints, e.g. `"r"(rbp)`, you might also be able to do `"%rbp"(rbp)` to really hammer it home.

Finally, I think the stack pointer should probably be an output.

Reply:
[115fd65](https://github.com/lifting-bits/remill/pull/625/commits/115fd655eb62aad6f18bb34a4a981409a3cebf0c)

Reply:
> In the register constraints, e.g. "r"(rbp), you might also be able to do "%rbp"(rbp) to really hammer it home.

I can't seem to find where that type of constraint (`%rbp`) is documented so I didn't go ahead with that. Addressed everything else though.